src/main/java/org/junitpioneer/internal/PioneerAnnotationUtils.java

```diff
@@ -204,18 +207,52 @@ private static boolean isContainerAnnotationOf(Annotation potentialContainer, Cl
 					.orElse(Collections.emptyList());
 	}
 
+	private static <A extends Annotation> List<A> findOnClass(Class<?> element, Class<A> annotationType,
+			boolean findRepeated, boolean findAllEnclosing) {
+		if (element == null || element == Object.class)
+			return Collections.emptyList();
+		if (findRepeated)
+			return AnnotationSupport.findRepeatableAnnotations(element, annotationType);
+
+		List<A> onElement = AnnotationSupport
+				.findAnnotation(element, annotationType)
+				.map(Collections::singletonList)
+				.orElse(Collections.emptyList());
+		List<A> onInterfaces = Arrays
```

Comment:
Hm. You always call `onClass` and now the variable is called `onInterface`.

Reply:
Will rename to `onType`